### PR TITLE
Added functional skeleton

### PR DESCRIPTION
**Pull request by Mingela**

### CODE REVIEW

File: src/main/java/iroha/validation/config/ValidationServiceContext.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+public class ValidationServiceContext {
+
+  private Collection<Validator> validators;
+  private TransactionProvider transactionProvider;
+  private TransactionSigner transactionSigner;
+  private KeyPair keyPair;
+
+  public ValidationServiceContext(
+      Collection<Validator> validators,
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/config/ValidationServiceContext.java]: Is it ok for these values to be null?

File: src/main/java/iroha/validation/config/ValidationServiceContext.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import iroha.validation.transactions.signatory.TransactionSigner;
+import iroha.validation.transactions.storage.TransactionProvider;
+import iroha.validation.validators.Validator;
+import java.security.KeyPair;
+import java.util.Collection;
+
+public class ValidationServiceContext {
+
+  private Collection<Validator> validators;
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/config/ValidationServiceContext.java]: we can make them all `final`

File: src/main/java/iroha/validation/rules/Rule.java [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
 public interface Rule {
 
+  /**
+   * Method for checking transaction rule satisfiability
+   *
+   * @param transaction Iroha proto transaction
+   * @return <code>true</code> if the transaction satisfies rule;
+   * <code>false</code> otherwise
+   */
+  Boolean isSatisfiedBy(TransactionOuterClass.Transaction transaction);
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/rules/Rule.java]: Why `Boolean`? Why not primitive `boolean`?

File: src/main/java/iroha/validation/service/impl/ValidationServiceImpl.java [hunk 4]
```diff
@@ -1,5 +1,39 @@
...
+import java.util.Collection;
 
+public class ValidationServiceImpl implements ValidationService {
+
+  private Collection<Validator> validators;
+  private TransactionProvider transactionProvider;
+  private TransactionSigner transactionSigner;
+  private KeyPair keyPair;
+
+  public ValidationServiceImpl(ValidationServiceContext validationServiceContext) {
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/service/impl/ValidationServiceImpl.java]: Can validationServiceContext be null? Looks like not

File: src/main/java/iroha/validation/validators/impl/SampleValidator.java [hunk 5]
```diff
@@ -1,5 +1,25 @@
...
+import iroha.validation.validators.Validator;
+import java.util.Collection;
+import java.util.Collections;
 
+public class SampleValidator implements Validator {
+
+  private Collection<Rule> rules;
+
+  public SampleValidator() {
+    this.rules = Collections.singletonList(new SampleRule());
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/validators/impl/SampleValidator.java]: we can move `Collections.singletonList(new SampleRule())` to 12th line. By doing that, we can remove this constructor.

File: src/main/java/iroha/validation/config/ValidationServiceContext.java (hunk 1 above)
Comment (Mingela) [reply src/main/java/iroha/validation/config/ValidationServiceContext.java]: A check has been added

File: src/main/java/iroha/validation/config/ValidationServiceContext.java (hunk 2 above)
Comment (Mingela) [reply src/main/java/iroha/validation/config/ValidationServiceContext.java]: Done

File: src/main/java/iroha/validation/rules/Rule.java (hunk 3 above)
Comment (Mingela) [reply src/main/java/iroha/validation/rules/Rule.java]: Done

File: src/main/java/iroha/validation/service/impl/ValidationServiceImpl.java (hunk 4 above)
Comment (Mingela) [reply src/main/java/iroha/validation/service/impl/ValidationServiceImpl.java]: A check has been added

File: src/main/java/iroha/validation/validators/impl/SampleValidator.java (hunk 5 above)
Comment (Mingela) [reply src/main/java/iroha/validation/validators/impl/SampleValidator.java]: The idea was to have several opportunities to instantiate this, added different constructors

File: src/main/java/iroha/validation/config/ValidationServiceContext.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  private final KeyPair keyPair;
+
+  public ValidationServiceContext(
+      Collection<Validator> validators,
+      TransactionProvider transactionProvider,
+      TransactionSigner transactionSigner,
+      KeyPair keyPair) {
+    Objects.requireNonNull(validators, "Validators collection must not be null");
+    // TODO rework isEmpty check during springify task
+    if(validators.isEmpty()) throw new RuntimeException("Validators collection must not be empty");
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/config/ValidationServiceContext.java]: I think it's better to throw `IllegalArgumentException`.

File: src/main/java/iroha/validation/validators/impl/SampleValidator.java [hunk 7]
```diff
@@ -1,5 +1,33 @@
...
+
+  public SampleValidator() {
+    this(new SampleRule());
+  }
+
+  public SampleValidator(Rule rule) {
+    this(Collections.singletonList(rule));
+  }
+
+  public SampleValidator(Collection<Rule> rules) {
```
Comment (dolgopolovwork) [inline src/main/java/iroha/validation/validators/impl/SampleValidator.java]: What if `rules` is null?

File: src/main/java/iroha/validation/config/ValidationServiceContext.java (hunk 6 above)
Comment (Mingela) [reply src/main/java/iroha/validation/config/ValidationServiceContext.java]: Done

File: src/main/java/iroha/validation/validators/impl/SampleValidator.java (hunk 7 above)
Comment (Mingela) [reply src/main/java/iroha/validation/validators/impl/SampleValidator.java]: A check has been added

**Pull request merged**